StortSpelProjekt/Engine/src/Renderer/TextManager.cpp

```diff
@@ -0,0 +1,274 @@
+	}
+}
+
+void TextManager::SetPos(float2 textPos, std::string name)
+{
+	bool exists = false;
+	for (auto data : m_TextDataMap)
+	{
+		if (data.first == name)
+		{
+			m_TextDataMap[name].pos = textPos;
+			exists = true;
+			break;
+		}
+	}
+
+	if (exists == false)
+	{
+		Log::PrintSeverity(Log::Severity::CRITICAL, "The text '%s', does not exist! Could not set position.\n", name.c_str());
+	}
+}
+
+void TextManager::SetScale(float2 scale, std::string name)
+{
+	bool exists = false;
+	for (auto data : m_TextDataMap)
+	{
+		if (data.first == name)
+		{
+			// Scale with the size of the window
+			Renderer* renderer = &Renderer::GetInstance();
+			HWND* hwnd = const_cast<HWND*>(renderer->m_pWindow->GetHwnd());
+			RECT rect;
+
+			float win_x = 0, win_y = 0;
+			if (GetWindowRect(*hwnd, &rect))
+			{
+				win_x = rect.right - rect.left;
+				win_y = rect.bottom - rect.top;
+			}
+
+			float scale_x = 0, scale_y = 0;
+			scale_x = win_x / 1000;
+			scale_y = win_y / 1000;
+			float aspect = scale_x / scale_y;
+
+			m_TextDataMap[name].scale.x = (scale.x * scale_x);
+			m_TextDataMap[name].scale.y = (scale.y * scale_y * aspect);
+
+			exists = true;
+			break;
+		}
+	}
+
+	if (exists == false)
+	{
+		Log::PrintSeverity(Log::Severity::CRITICAL, "The text '%s', does not exist! Could not set scale.\n", name.c_str());
+	}
+}
+
+void TextManager::SetPadding(float2 padding, std::string name)
+{
+	bool exists = false;
+	for (auto data : m_TextDataMap)
+	{
+		if (data.first == name)
+		{
+			m_TextDataMap[name].padding = padding;
+			exists = true;
+			break;
+		}
+	}
+
+	if (exists == false)
+	{
+		Log::PrintSeverity(Log::Severity::CRITICAL, "The text '%s', does not exist! Could not set padding.\n", name.c_str());
+	}
+}
+
+void TextManager::SetColor(float4 color, std::string name)
+{
+	bool exists = false;
+	for (auto data : m_TextDataMap)
+	{
+		if (data.first == name)
+		{
+			m_TextDataMap[name].color = color;
+			exists = true;
+			break;
+		}
+	}
+
+	if (exists == false)
+	{
+		Log::PrintSeverity(Log::Severity::CRITICAL, "The text '%s', does not exist! Could not set color.\n", name.c_str());
+	}
+}
+
+Font* TextManager::GetFont() const
+{
+	return m_pFont;
+}
+
+Texture* TextManager::GetFontTexture() const
+{
+	return m_pFont->GetTexture();
+}
+
+Text* TextManager::GetText(std::string name)
+{
+	return m_TextMap[name];
+}
+
+std::map<std::string, Text*>* TextManager::GetTextMap()
+{
+	return &m_TextMap;
+}
+
+const int TextManager::GetNumOfTexts() const
+{
+	return m_TextMap.size();
+}
+
+const int TextManager::GetNumOfCharacters(std::string name)
+{
+	return m_TextDataMap[name].text.size();
+}
+
+void TextManager::submitText(Text* text, std::string name)
+{
+	m_TextMap.insert({ name, text });
+}
+
+void TextManager::replaceText(Text* text, std::string name)
+{
+	bool found = false;
+	for (auto textMap : m_TextMap)
+	{
+		if (textMap.first == name)
+		{
+			delete m_TextMap[name];
+			m_TextMap[name] = text;
+			found = true;
+		}
+	}
```

Comment:
By now I lost count of how many times you loop. Maybe you should really, REALLY change the data structure of the TextMap.